### ISSUE TITLE
Operator overloading regression

### ISSUE THREAD

**Issue by RealyUniqueName** [HaxeFoundation/haxe#8790]
```haxe
class Main {
	public static function main() {
		$type(1 + new Int8(1));
	}
}

abstract Int8(Int) {

	public inline function new(value:Int) {
		this = value;
	}

	@:op(A + B) inline function addition(b:Int8):Int8 {
		throw 'oops';
	}

	@:op(A + B) static function intAddition(a:Int8, b:Int):Int {
		return b + cast a;
	}
}
```
This should not compile because `Int + Int8` is not defined in any `@:op`.
But Haxe 4.0 picks `addition` and compiles without errors:
```
$ haxe build.hxml 
src/Main.hx:3: characters 9-24 : Warning : Int8
```
Output with Haxe 3.4:
```
$ haxe build.hxml 
src/Main.hx:3: characters 12-23 : Int8 should be Int
src/Main.hx:3: characters 8-23 : Warning : Int
```

**Comment by Ilir-Liburn** [comment]
I'm getting: haxe.lang.HaxeException: oops (CS target)

One thing I don't understand: you say Int8 + Int is not defined in any @:op, but you have intAddition(a:Int8, b:Int). And it is not commutative in your example, so how to expect Int + Int8 to work through abstract?

**Comment by Ilir-Liburn** [comment]
This is working

```haxe
abstract Int8(Int) {

	public inline function new(value:Int) {
		this = value;
	}

    @:commutative
	@:op(A + B) static function intAddition(a:Int8, b:Int):Int {
		return b + cast a;
	}

	@:op(A + B) inline function addition(b:Int8):Int8 {
		throw 'oops';
	}
}

      $type(1 + new Int8(1));
```

EDIT: however, I don't understand why is addition() called if it is placed before intAddition(). Shouldn't addition() be called for Int8 + Int8, not for Int + Int8?

**Comment by RealyUniqueName** [comment]
> One thing I don't understand: you say Int8 + Int is not defined in any @:op, but you have intAddition(a:Int8, b:Int).

Yeah, my bad. I meant `Int + Int8` is not defined.

**Comment by RealyUniqueName** [comment]
Reverting f1e00acb633835538bd3c89da299684ee81b9734 fixes the regression.

**Comment by RealyUniqueName** [comment]
Leaving open until merged into 4.0_bugfix
